### DISCUSSION TITLE
How to access editor DOM in react ?

### DISCUSSION THREAD

**Discussion by ShenQingchuan** [Milkdown/milkdown#1151] [Q&A] [answered]
I copied and refactor a little bit from the official site's playground. 
My problem is about the `link-widget` component, i.e. the input box of the link element.

I found that the behaviour is somewhat not as users' expectation when I hit `Enter` key on focusing in the link-widget's input box, there's a new line inserted into editor, but what I want is to exit the input box and my cursor goes back to the position right after this link element.

Is there any best practice for me to access the DOM or do something else to implement the behaviour I talked above?

**Comment by Saul-Mirone** [comment]
I think if you just want to prevent some keymap of editor when focusing input widget, have you tried `event.stopPropagation()`?

**Reply by ShenQingchuan** [reply]
Yeah! But this is the first step! after `e.preventDefault()` and `e.stopPropagation()`, I'd like to put my cursor back to the position where I clicked, but for now when I hit Enter nothing happened because I have not written any code about it.

So my question here is how to access the editor DOM itself and re-focus it.

**Reply by ShenQingchuan** [reply]
https://github.com/Milkdown/milkdown/assets/46062972/8ee64b05-fb70-48a6-80a5-790fc49895e2

I made a video record for you and you can see my action on Milkdown playground site.

I hit enter and the link got splitted, I think this is not a user-expected behaviour. It should exit that input box and go back to a normal state in editor.

**Reply by s524797336** [reply]
You can use transication.setSelection api to modify selection in milkdown

**Reply by ShenQingchuan** [reply]
Sure! That's a good idea, but I indeed want to know about how to access these ProseMirror primitive concept context in Milkdown?

**Reply by Saul-Mirone** [reply, marked as answer]
Okay, I got it. Here's how you can get edtior's state and view:

```ts
import { editorViewCtx } from '@milkdown/core';

editor().action((ctx) => {
    const view = ctx.get(editorViewCtx);
    const state = view.state;
    const selection = state.selection;
    const { $to } = selection;
    const to = $to.after();
    const $sel = TextSelection.near(state.doc.resolve(to));
    view.dispatch(state.tr.setSelection($sel));
    view.focus();
})
```

**Reply by ShenQingchuan** [reply]
it works! Thanks a lot!
Milkdown is awesome!!!
